SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -36,6 +36,17 @@ HGCDigitizerBase<DFr>::HGCDigitizerBase(const edm::ParameterSet& ps) {
   edm::ParameterSet feCfg = myCfg_.getParameter<edm::ParameterSet>("feCfg");
   myFEelectronics_ = std::unique_ptr<HGCFEElectronics<DFr>>(new HGCFEElectronics<DFr>(feCfg));
   myFEelectronics_->SetNoiseValues(noise_fC_);
+  Dim_ = noise_fC_.size();
+  GenGaussianNoise(NoiseMean_, NoiseStd_);
+}
+
+template <class DFr>
+void HGCDigitizerBase<DFr>::GenGaussianNoise(const double NoiseMean, const double NoiseStd) {
+  for (size_t i = 0; i < NoiseArrayLength_; i++) {
+    for (int j = 0; j < samplesize_; j++) {
+      GaussianNoiseArray_[i][j] = CLHEP::RandGaussQ::shoot(NoiseMean, NoiseStd);
```

Comment:
(general question) Should the random number seeds be controlled by the RandomNumberGeneratorService?

Reply:
Yes, this will not be reproducible.

Reply:
@adas1994 this needs to be addressed before the PR can be considered for integration